pkg/operator/operator.go

```diff
@@ -209,6 +230,40 @@ func isOwned(obj interface{}) (bool, error) {
 	return ok, nil
 }
 
+func (optr *Operator) eventHandlerSingleton() cache.FilteringResourceEventHandler {
+	workQueueKey := fmt.Sprintf("%s/%s", optr.namespace, optr.name)
+	addToQueue := func(obj interface{}) {
+		glog.V(4).Infof("Event: Add")
+		logResource(obj)
+		optr.queue.Add(workQueueKey)
+	}
+
+	return cache.FilteringResourceEventHandler{
+		FilterFunc: isMachineWebhook,
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc:    addToQueue,
+			DeleteFunc: addToQueue,
+			UpdateFunc: func(old, new interface{}) {
+				addToQueue(new)
+			},
+		},
+	}
+}
+
+func isMachineWebhook(obj interface{}) bool {
+	mutatingWebhook, ok := obj.(admissionregistrationv1.MutatingWebhookConfiguration)
+	if ok {
+		return mutatingWebhook.Name == "machine-api"
+	}
+
+	validatingWebhook, ok := obj.(admissionregistrationv1.MutatingWebhookConfiguration)
+	if ok {
+		return validatingWebhook.Name == "machine-api"
+	}
```

Comment:
Would the objects coming through this not be pointers?

Reply:
That's true, thanks for noticing this. Although with one long runnig controller thread, the events here are still blocked...